Bottom: c7118b24e57cf9dadbc32d3e46fb56a74907a0d8
Top:    3f276479a3775815b424d3d56fba03ea5e295f10
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-03 18:03:24 +0100

hid/gtk: Always show the PCB name on the title-bar, not the toolbar

This saves space and reduces code complexity


---

diff --git a/src/hid/gtk/gui-config.c b/src/hid/gtk/gui-config.c
index cbd46a4..d29d485 100644
--- a/src/hid/gtk/gui-config.c
+++ b/src/hid/gtk/gui-config.c
@@ -122,7 +122,6 @@ static gchar *config_dir, *color_dir;
 static ConfigAttribute config_attributes[] = {
   {"gui-compact-horizontal", CONFIG_Boolean, &_ghidgui.compact_horizontal},
   {"gui-compact-vertical", CONFIG_Boolean, &_ghidgui.compact_vertical},
-  {"gui-title-window", CONFIG_Boolean, &_ghidgui.ghid_title_window},
   {"use-command-window", CONFIG_Boolean, &_ghidgui.use_command_window},
   {"save-in-tmp", CONFIG_Unused, NULL},
   {"grid-units", CONFIG_Unused, NULL},
@@ -859,16 +858,6 @@ config_compact_vertical_toggle_cb (GtkToggleButton * button, gpointer data)
 }
 
 static void
-config_title_window_cb (GtkToggleButton * button, gpointer data)
-{
-  gboolean active = gtk_toggle_button_get_active (button);
-
-  ghidgui->ghid_title_window = active;
-  ghid_window_set_name_label (ghidgui->name_label_string);
-  ghidgui->config_modified = TRUE;
-}
-
-static void
 config_general_toggle_cb (GtkToggleButton * button, void * setting)
 {
   *(gint *)setting = gtk_toggle_button_get_active (button);
@@ -922,11 +911,6 @@ config_general_tab_create (GtkWidget * tab_vbox)
 			       config_compact_vertical_toggle_cb, NULL,
 			       _("Alternate window layout to allow smaller vertical size"));
 
-  ghid_check_button_connected (vbox, NULL, ghidgui->ghid_title_window,
-			       TRUE, FALSE, FALSE, 2,
-			       config_title_window_cb, NULL,
-			       _("Put layout name on the window title bar"));
-
   vbox = ghid_category_vbox (tab_vbox, _("Backups"), 4, 2, TRUE, TRUE);
   ghid_check_button_connected (vbox, NULL, Settings.SaveInTMP,
 			       TRUE, FALSE, FALSE, 2,
diff --git a/src/hid/gtk/gui-top-window.c b/src/hid/gtk/gui-top-window.c
index fb5a32a..7ed1d77 100644
--- a/src/hid/gtk/gui-top-window.c
+++ b/src/hid/gtk/gui-top-window.c
@@ -626,29 +626,13 @@ ghid_window_set_name_label (gchar * name)
   if (!ghidgui->name_label_string || !*ghidgui->name_label_string)
     ghidgui->name_label_string = g_strdup (_("Unnamed"));
 
-  if (!ghidgui->name_label)
-    return;
-
   if (!PCB->Filename  || !*PCB->Filename)
     filename = g_strdup(_("Unsaved.pcb"));
   else
     filename = g_strdup(PCB->Filename);
 
-  if (ghidgui->ghid_title_window)
-    {
-      gtk_widget_hide (ghidgui->name_label);
-      str = g_strdup_printf ("%s%s (%s) - PCB", PCB->Changed ? "*": "",
-                             ghidgui->name_label_string, filename);
-    }
-  else
-    {
-      gtk_widget_show (ghidgui->name_label);
-      str = g_strdup_printf (" <b><big>%s</big></b> ",
-                             ghidgui->name_label_string);
-      gtk_label_set_markup (GTK_LABEL (ghidgui->name_label), str);
-      str = g_strdup_printf ("%s%s - PCB", PCB->Changed ? "*": "",
-                             filename);
-    }
+  str = g_strdup_printf ("%s%s (%s) - PCB", PCB->Changed ? "*": "",
+                         ghidgui->name_label_string, filename);
   gtk_window_set_title (GTK_WINDOW (gport->top_window), str);
   g_free (str);
   g_free (filename);
@@ -1210,7 +1194,6 @@ ghid_build_pcb_top_window (void)
   GtkWidget *vbox, *frame;
   GtkWidget *label;
   GHidPort *port = &ghid_port;
-  gchar *s;
   GtkWidget *scrolled;
 
   window = gport->top_window;
@@ -1266,20 +1249,6 @@ ghid_build_pcb_top_window (void)
                     FALSE, FALSE, 0);
 
   /*
-   * The board name is optionally in compact_vbox and the position
-   * labels will be packed below or to the side.
-   */
-  ghidgui->name_label = gtk_label_new ("");
-  gtk_label_set_use_markup (GTK_LABEL (ghidgui->name_label), TRUE);
-  if (ghidgui->name_label_string)
-    s = g_strdup_printf (" <b><big>%s</big></b> ", ghidgui->name_label_string);
-  else
-    s = g_strdup ("<b><big>%s</big></b>");
-  gtk_label_set_markup (GTK_LABEL (ghidgui->name_label), s);
-  g_free (s);
-  gtk_box_pack_start (GTK_BOX (ghidgui->compact_vbox), ghidgui->name_label, TRUE, FALSE, 6);
-
-  /*
    * The position_box pack location depends on user setting of
    * compact horizontal mode.
    */
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 1cb3865..8c6478f 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -96,7 +96,7 @@ typedef struct
   GtkActionGroup *main_actions,
     *change_selected_actions, *displayed_name_actions;
 
-  GtkWidget *name_label,
+  GtkWidget
     *status_line_label,
     *cursor_position_relative_label,
     *cursor_position_absolute_label,
@@ -126,7 +126,8 @@ typedef struct
     small_label_markup,
     compact_horizontal,
     compact_vertical,
-    ghid_title_window, use_command_window, creating;
+    use_command_window,
+    creating;
 
   gint n_mode_button_columns,
     top_window_width,
